### PR TITLE
chore(ci): Unify workflow names

### LINKED ISSUE #13275: Clean up GitHub Actions workflow names
Following on from https://github.com/magma/magma/pull/13197, I think the names of the workflows should be unified. In particular, looking at https://github.com/magma/magma/actions, some workflows have descriptive names, e.g. "Push Bazel Cache To S3", while some just use the name of the workflow file, e.g. "agw-build". This means that the "Workflows" section on the left of the Actions page has two ordered lists for the workflows, which can make finding a particular workflow harder than it should be, particularly for new users. I would suggest we use descriptive names for all of them.

See https://github.com/magma/magma/issues/13275#issuecomment-1182972207 for proposed naming conventions!

#### :warning: Beware of the following!
- Changing the workflow names might impact the required workflows for PRs.
- Mind CI dashboard need adoptions.
- Changing the workflow `name:` will preserve the run history, but only if you do not change the file name, i.e. leave the file names as is first.

Tasks
- [x] Change the names according to {component} {verb} [{addition}] (e.g. AGW build with bazel) so that component jobs are automatically grouped in alphabetic sorting. Before merging the pull request, discuss with Max to make sure required job names are changed in GitHub.
- [x] Document the naming scheme in the Wiki.
- [x] Prepare internal review.

### PR DESCRIPTION
**Pull request by voisey**
## Summary

Closes https://github.com/magma/magma/issues/13275.

Unifies the names of the workflows by giving them non-default names (e.g. fossa-workflow -> Orc8r Analyze With Fossa) and by having the names lead with the component they relate to (e.g. insync-check -> Orc8r Check Generated Files In Sync). This PR should make it easier to navigate through the actions and easier to understand what the workflows actually do.

:warning: When merging, we need to propagate these changes to the configuration of the required workflows. CC @maxhbr 
-> Update: It was not necessary to do this because the required jobs are all steps (which are not updated here), not workflows.

## Test Plan

- CI

## Suggestion for formatting which will be added to the Wiki upon merging this:

The convention for naming workflows is as follows:

**{component} {verbs} {additions}**

with

- {component} in AGW, CWAG, Docs, DP, FeG, Magma, NMS, Orc8r, PR
- {verbs} in Analyse, Backport, Build, Check, Deploy, Format, Generate, Lint, Promote, Publish, Test
- {addition} being a short descriptive addition 

Rules and notes:
- All words in the name should be capitalized.
- {component}:
  - Exactly one component should be chosen for every workflow.
  - “Magma” is intended for workflows relating to multiple Magma components.
  - “PR” is intended for workflows that interact with pull requests.
- {verbs}:
  - In the case that multiple verbs are applicable, the last separator should be an ampersand enclosed in one space each side, any other separator should be a comma followed by one space, e.g. `Build & Test`, `Lint, Build & Test`

### CODE REVIEW

**Review by Neudrino — CHANGES_REQUESTED** [review]
- Was there a discussion and decision on the naming scheme (https://github.com/magma/magma/issues/13275#issuecomment-1182972207)?
- Has the naming scheme been documented in the contributor guide?
- Has a justification been documented, toward why one would use the selected naming scheme (over the other(s))?
- A rule for workflows not belonging to any component has not been laid out.
- The above should be visibly linked in the PR.
- I cannot see consistent adherence to any naming scheme.
- For variable names most developers consider it an anti-pattern to include the type into the name, e.g. `List plantList = [oak, beech]`. The same applies here, having a workflow called "workflow" does not add any information and should be avoided.

:warning: If we change the naming, we do probably not want to go in multiple iteration, but get it right the first time, as we have to adopt all PR checks manually!

**Comment by sebathomas** [comment]
We agreed on the basic pattern of component-first instead of action-first because most people felt it would be more useful to group that way (so that e.g. all AGW jobs are together and not all Test jobs).

In general, any scheme will have the problem that the different workflow jobs are not cut in the same way, and do many different things at the same time (build, test, lint, deploy in various combinations). So I think it's better to have the discussion on this PR with the concrete workflow steps in mind, instead of trying to decide on a general scheme first.

I agree it would be good to come up with better names for the "workflow" workflows. Suggestions are very welcome.

How to handle workflows that don't fit a component is not so easy either. The solution at the moment groups workflows that touch multiple components in "Magma" and workflows that are related to the concrete PR (like the DCO check) in "PR". There might be other ideas like that.

**Comment by nstng** [comment]
Just mentioning an inconsistency - might be out of scope here and worth a followup pull request:
* in the trigger config `on: workflow_run: workflows:` it seems workflow names or workflow file names can be used (but I did not find this in the documentation by a quick look)
* It looks like only in `comment-pr-on-check-failure.yml` the workflow names are used.
* I assume name and file name are subjected to be changed and possible trigger might be forgotten - but the file name feels more consistent.

Should this be aligned?

**Comment by Neudrino** [comment]
Excellent finding, @nstng. I agree, that should probably be aligned. :heavy_plus_sign: for file names.

As for the `comment-pr-on-check-failure.yml`, I believe this should really be replaced by a proper implementation (e.g. https://github.com/magma/magma/pull/13607 or abstraction over it). However, this would clearly be out of scope of the ticket.

File: .github/workflows/cwf-operator.yml
Comment (Neudrino) [inline .github/workflows/cwf-operator.yml:12]: Is CWAG really the same as cwf?

File: .github/workflows/deploy-build-from-pr.yml
Comment (Neudrino) [inline .github/workflows/deploy-build-from-pr.yml:12]: Just making sure that the name change is intended here, in terms of that there is nothing deployed, but only published.

File: .github/workflows/golang-build-test.yml
Comment (Neudrino) [inline .github/workflows/golang-build-test.yml]: `AGW Build & Test Experimental Go Code`?
And cut the exception for this part from the definition above.

File: .github/workflows/lte-integ-test-bazel-magma-deb.yml
Comment (Neudrino) [inline .github/workflows/lte-integ-test-bazel-magma-deb.yml]: The `LTE` is probably not necessary, if we have the AGW prefix. (Also below.)

File: .github/workflows/lte-integ-test-containerized.yml
Comment (Neudrino) [inline .github/workflows/lte-integ-test-containerized.yml]: Chose this as arbitrary example:
Does it make sense to exclude some worlds from the description? Or should they be allowed?
E.g. `AGW Test LTE Integration With Make On Containerized Build`

In case they should not be used, a rule in the rules section should be formulated.

File: .github/workflows/reviewdog-workflow.yml
Comment (Neudrino) [inline .github/workflows/reviewdog-workflow.yml]: The Reviewdog workflow is only a `Lint` type of workflow, imho.

File: .github/workflows/semantic-pr.yml
Comment (Neudrino) [inline .github/workflows/semantic-pr.yml]: We should check again if this is just checking the PR title, or also the commit message. In case a commit message is also checked, the name should be adopted to make this clearer and not give a false impression.

File: .github/workflows/sudo-python-tests.yml
Comment (Neudrino) [inline .github/workflows/sudo-python-tests.yml:12]: Those are integration tests, right?
In that case the naming scheme from above would apply. E.g. `AGW Test Integration Sudo Python`.

**Review by Neudrino — COMMENTED** [review]
The names are sometimes with quotes and without. This should be made consistent. If quotes are not necessary, I would go for the form _without_ quotes.

File: .github/workflows/reviewdog-workflow.yml
Comment (voisey) [reply .github/workflows/reviewdog-workflow.yml]: Good point. Fixed

File: .github/workflows/golang-build-test.yml
Comment (voisey) [reply .github/workflows/golang-build-test.yml]: Agreed, that's better. I've done both

File: .github/workflows/fossa-workflow.yml
Comment (voisey) [reply .github/workflows/fossa-workflow.yml]: Fixed

File: .github/workflows/codeql-analysis.yml
Comment (voisey) [reply .github/workflows/codeql-analysis.yml]: Fixed

**Comment by voisey** [comment]
> The names are sometimes with quotes and without. This should be made consistent. If quotes are not necessary, I would go for the form _without_ quotes.

Thanks for the suggestion. I've removed the quotes

File: .github/workflows/sudo-python-tests.yml
Comment (voisey) [reply .github/workflows/sudo-python-tests.yml:12]: My understanding is that despite them being run in the integration test workflow in the past, they're not actually integration tests. Can you please confirm @nstng?

Comment (nstng) [reply .github/workflows/sudo-python-tests.yml:12]: They are kind of in a gray zone - unit tests that need a running ovs. IMO I would not call them integration tests because they do not need magma services - I think `AGW Test Sudo Python` is a good name.

File: .github/workflows/semantic-pr.yml
Comment (voisey) [reply .github/workflows/semantic-pr.yml]: You're right. It checks the commit message if it's a PR with only one commit. What do you think is best here? "PR Check Title Or Commit Message"?

File: .github/workflows/lte-integ-test-bazel-magma-deb.yml
Comment (voisey) [reply .github/workflows/lte-integ-test-bazel-magma-deb.yml]: I thought these test the LTE, i.e. 4G, capabilities of the AGW, which supports 5G too. If so, I think the inclusion in the name could be important?

**Review by LKreutzer — COMMENTED** [review]
Would it make sense to also replace all occurrences of the workflow names inside the workflows (e.g. Slack notifications)? This could be done without redundancy by using the `"${{ github.workflow }}"` object, which is just the value of the `name:` field.

**Comment by jheidbrink** [comment]
Was there a discussion about changing filenames? For some workflows there was a kind of mapping `filename <--> workflow name`, and by changing the workflow names and not the filenames we are moving away from that mapping.

**Comment by sebathomas** [comment]
From what I understand, by not changing filenames we keep the Github Actions history (so all previous runs of the workflow will still be available under the new name).

**Comment by jheidbrink** [comment]
Do you know if we can preserve history by first only changing workflow names and in a second step only the filenames?

**Comment by LKreutzer** [comment]
> Do you know if we can preserve history by first only changing workflow names and in a second step only the filenames?

Seems to me that if you rename the file and keep the `name:` the same then the workflow runs in the actions tab all vanish.

File: .github/workflows/cwf-operator.yml
Comment (voisey) [reply .github/workflows/cwf-operator.yml:12]: I suppose CWF is broader than CWAG because the former refers to the WiFi while the latter refers to the gateway only. I think in the case of the integration test workflow `cwf-integ-test.yaml`, using CWAG as the component makes sense in analogy with AGW for the LTE integ tests. However, perhaps for the operator, CWF does make sense. The operator is referred to as the "CWF Operator" [here](https://github.com/magma/magma/blob/master/cwf/k8s/cwf_operator/README.md), for example. Any thoughts on this? CC @sebathomas @crasu

**Comment by jheidbrink** [comment]
Ok, so it might be better to just keep the filenames then.

File: .github/workflows/cwf-operator.yml
Comment (sebathomas) [reply .github/workflows/cwf-operator.yml:12]: Isn't the operator just running things on the gateway (or basically running the gateway)?

In any case, we should have everything related to carrier wifi under a single top-level component name. Maybe CWF would be better than CWAG but I have no strong preference.

File: .github/workflows/deploy-build-from-pr.yml
Comment (voisey) [reply .github/workflows/deploy-build-from-pr.yml:12]: Yes, I would say that this is just publishing.

File: .github/workflows/lte-integ-test-containerized.yml
Comment (voisey) [reply .github/workflows/lte-integ-test-containerized.yml]: I've added the "With" to all of these workflows but not the "On" because with the "On" it makes it sound like the integration tests are being run with Make/Bazel, when in fact its the build that is done with Make/Bazel.

**Comment by voisey** [comment]
> Would it make sense to also replace all occurrences of the workflow names inside the workflows (e.g. Slack notifications)? This could be done without redundancy by using the `"${{ github.workflow }}"` object, which is just the value of the `name:` field.

Nice suggestion. I would suggest keeping this for a follow-up PR to keep this one at a reasonable size.

File: .github/workflows/cwf-operator.yml
Comment (crasu) [reply .github/workflows/cwf-operator.yml:12]: For me both is fine it should just be consistent

Comment (voisey) [reply .github/workflows/cwf-operator.yml:12]: Okay, thanks, then I propose we stick with CWAG

**Comment by voisey** [comment]
@Neudrino I believe I have addressed all of your comments, so if you could take another look, that would be great!

File: .github/workflows/federated-integ-test.yml
Comment (Neudrino) [inline .github/workflows/federated-integ-test.yml:12]: Just noticed, might this be rather in the category `FeG`, if it is specific to this component?

Comment (sebathomas) [reply .github/workflows/federated-integ-test.yml:12]: I think the idea to make it "Magma" was because these are tests of AGW, Orc8r and FeG working together.

**Review by Neudrino — APPROVED** [review]
Could you publish the Guidelines already in the Contributors Guide and link the addition here please. This would make it easier for efforts as e.g. https://github.com/magma/magma/pull/14127, where new workflows are added.

**Review by mpfirrmann — APPROVED** [review]

**Comment by voisey** [comment]
@Neudrino I've added a slightly modified version of the guidelines here: https://github.com/magma/magma/wiki/Contributing-Code#naming-conventions

Note that the names of the workflows in the two tables [here](https://github.com/magma/magma/wiki/Contributing-Code#naming-conventions) in the Wiki will also need to be changed to reflect the changes in this PR, but I would prefer to do this once the PR has been approved by all the approver groups.

**Review by sebathomas — APPROVED** [review]
:+1:

**Review by LKreutzer — APPROVED** [review]
Lgtm

If we want these rules enforced we should create a check on PRs that create workflows or rename existing ones.

**Comment by Neudrino** [comment]
> Note that the names of the workflows in the two tables ~~[here](https://github.com/magma/magma/wiki/Contributing-Code#naming-conventions)~~ [[really here](https://github.com/magma/magma/wiki/Contributing-Code#continuous-integration-ci--continuous-deployment-cd)] in the Wiki will also need to be changed to reflect the changes in this PR, but I would prefer to do this once the PR has been approved by all the approver groups.

We should discuss to remove this list, as it is a manually curated list of the configured CI checks, which is unnecessary duplication and very prone to get out of date, as someone changes the settings. If at all a link to the respective configuration page should be given (even though that page is only visible to repository administrators).

**Pull request merged**
